pkg/operator/ceph/cluster/status/conditions.go

```diff
@@ -0,0 +1,300 @@
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: message,
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionUpdating(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionReady,
+		Status:  v1.ConditionFalse,
+		Reason:  "ClusterNotReady",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionAvailable(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionInitialize(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnecting,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionConnected,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionFailure,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionIgnored,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionUpgrading,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionAvailable,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionExpanding,
+		Status:  v1.ConditionFalse,
+		Reason:  "",
+		Message: "",
+	})
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionNotReady,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+func ConditionExpanding(context *clusterd.Context, conditions *[]cephv1.RookConditions, reason, message string) {
+	setCondition(context, conditions, cephv1.RookConditions{
+		Type:    cephv1.ConditionExpanding,
+		Status:  v1.ConditionTrue,
+		Reason:  reason,
+		Message: message,
+	})
+}
+
+// applying the latest condition on to the cluster CRD file
+func applyCondition(context *clusterd.Context, conditions *[]cephv1.RookConditions, newCondition cephv1.RookConditions) {
+	if name == "" {
```

Comment:
where are getting `name` from?

Reply:
There is a function named `ClusterInfo`. So this function will get the `name` and `namespace` of the rook-cluster.